scripts/playground/template/app/models/user.server.ts

```diff
@@ -5,11 +5,11 @@ import { prisma } from "~/db.server";
 
 export type { User } from "@prisma/client";
 
-export async function getUserById(id: User["id"]) {
```

Comment:
This is copy/pasted from our `indie-stack`, so I would leave it in sync tbh

Don't know if we should change it there though, as I personally like the fact that you can see straight away that a function is async or not (Prisma returns promises)

Reply:
> in sync

Heh, I would suggest enabling this rule in `indie-stack` then. Then they'd stay in sync!

> is async or not

But the type signature exposed to end users is the same, no? `(id: User["id"]) => Promise<User>`? 

Aside: it is an interesting, subtle difference - being "async" vs. synchronously returning a Promise... I'd posit that it's even a little misleading to see a function being marked as `async` but then not itself including awaiting code. 

<details>
<summary>Details on why <code>return await</code> can improve call stacks, for any curious readers</summary>

If a `try`/`catch` wraps an `await`, only functions that have an `await` inside will be included in the call stack. Running snippet demonstrates different call stacks:

```js
async function inside() {
  await new Promise(setTimeout);
  throw new Error("Oh no!");
}

async function middle(useAwait) {
  return useAwait ? await inside() : inside();
}

async function outer(useAwait) {
  try {
    await middle(useAwait);
  } catch (error) {
    return error;
  }
}

console.log("Using await:", await outer(true));
console.log("Without await:", await outer(false));
```

Output in Node:

```plaintext
$ node temp.mjs
Using await: Error: Oh no!
    at inside (file:///path/to/temp.mjs:3:9)
    at async middle (file:///path/to/temp.mjs:7:21)
    at async outer (file:///path/to/temp.mjs:12:5)
    at async file:///path/to/temp.mjs:18:29
Without await: Error: Oh no!
    at inside (file:///path/to/temp.mjs:3:9)
    at async outer (file:///path/to/temp.mjs:12:5)
    at async file:///path/to/temp.mjs:19:31
```

</details>

Ref: https://typescript-eslint.io/rules/return-await

If you'd really prefer this be `async`, what do you think about having this `return await prisma.user.findUnique`? That way the `getUserById` function is included in call stacks in case of throwing an error.

Reply:
The signature will be the same, so there's no problem in that day
It's just that looking at the code you immediately see if a func is async or not without needing to know if Prisma returns a Promise

I don't like awaiting a promise you return either tbh 🙈

But again: this is my personal taste and I'm happy to follow whatever the team decides here and just get used to it 🤷‍♂️

Reply:
I think we want to mostly just take whatever the preset/recommended configs include and avoid any type of bikeshedding over eslint rules/prefs.  Since `@typescript-eslint/recommended-type-checked` incudes the rule I vote we make it pass here and (eventually when we re-work our eslint configs) in the stack